plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,658 @@
+
+    result = {}
+    object_info = {}
+
+    try:
+        object_info = connection.get_object(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        pass
+
+    if len(object_info) != 0:
+        # Remove Body, ResponseMetadata from object_info, convert to snake_case
+        del(object_info['Body'])
+        del(object_info['ResponseMetadata'])
+        object_info = camel_dict_to_snake_dict(object_info)
+
+    result['object_data'] = object_info
+
+    return result
+
+
+@AWSRetry.jittered_backoff(retries=10)
+def _list_bucket_objects(connection, **params):
+    paginator = connection.get_paginator('list_objects')
+    return paginator.paginate(**params).build_full_result()
+
+
+def list_bucket_objects(connection, module, bucket_name):
+    params = {}
+    params['Bucket'] = bucket_name
+
+    result = []
+    list_objects_response = {}
+
+    try:
+        list_objects_response = _list_bucket_objects(connection, **params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        pass
+
+    if len(list_objects_response) != 0:
+        # convert to snake_case
+        for response_list_item in list_objects_response['Contents']:
+            result.append(response_list_item['Key'])
+
+    module.exit_json(s3_keys=result)
+
+
+def main():
+
+    argument_spec = dict(
+        object_details=dict(type='dict', options=dict(
+            object_acl=dict(type='bool', default=False),
+            object_legal_hold=dict(type='bool', default=False),
+            object_lock_configuration=dict(type='bool', default=False),
+            object_retention=dict(type='bool', default=False),
+            object_tagging=dict(type='bool', default=False),
+            object_attributes=dict(type='bool', default=False),
+            attributes_list=dict(type='list', elements='str', choices=['ETag', 'Checksum', 'ObjectParts', 'StorageClass', 'ObjectSize']),
+        )),
+        bucket_name=dict(required=True, type='str'),
+        object_name=dict(type='str'),
+    )
+
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+    )
+
+    try:
+        connection = module.client('s3', retry_decorator=AWSRetry.jittered_backoff())
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        module.fail_json_aws(e, msg='Failed to connect to AWS')
+
+    result = []
+
+    bucket_name = module.params.get('bucket_name')
+    object_name = module.params.get('object_name')
+    requested_details = module.params.get('object_details')
+
+    if requested_details and len(requested_details['attributes_list']) != 0:
+        module.require_botocore_at_least('1.24.7', reason='required for s3.get_object_attributes')
+        if not requested_details['attributes_list']:
+            module.fail_json(msg='Please provide attributes_list list to retrieve s3 object_attributes.')
+
+    if requested_details:
```

Comment:
What is requested_details are specified and no object_name is provided? I would expect to have requested_details for every object in the bucket. Am I wrong?

@jillr @tremble

Reply:
@alinabuzachis thank you! good catch, added code modifications required.
The module now will return a list of dicts containing requested details for all the objects in a bucket when object name is not specified.

sample task

```
- name: Retrieve detailed S3 information for all objects in the bucket
  amazon.aws.s3_object_info:
    bucket_name: MyTestBucket
    object_details:
      object_acl: true
      object_attributes: true
      attributes_list:
        - ETag
        - ObjectSize
        - StorageClass
```